### ISSUE TITLE
Server grab focus on start in Windows

### ISSUE THREAD

**Issue by Chairn** [ddnet/ddnet#5526]
Reported by Voxel on Discord, louis says it has always been like that for him.

Doesn't happen on Linux for me.

**Comment by Robyt3** [comment]
That just how Windows works. Dealing with focus stealing has been an unsolved issue for a long time. If you run an application from explorer it will give that application focus. From what I read, the current version of PowerToys/TweakUI cannot change the focus stealing behavior anymore, but it was previously possible to do so, though it would probably affect all windows opened with explorer. If you want to start a server without giving the console window focus, you need to run it as a service or with autorun.

**Comment by Chairn** [comment]
Doesn't take focus on win 7, but i couldn't try from explorer as it starts too fast before i can tab out

**Comment by Robyt3** [comment]
> No flags to start it minimized? windows defs has that in rightclick->settings for .exe

Indeed. You can create a shortcut to the server exe and then select `Run` = `Minimized` in the shortcut's properties dialog, which will start the server with initially minimized console window.

**Comment by heinrich5991** [comment]
There's the creation flag `CREATE_NO_WINDOW` (https://docs.microsoft.com/en-us/windows/win32/procthread/process-creation-flags):
> The process is a console application that is being run without a console window. Therefore, the console handle for the application is not set.
This flag is ignored if the application is not a console application, or if it is used with either `CREATE_NEW_CONSOLE` or `DETACHED_PROCESS`.

Alternatively, the stuff mentioned by @Jupeyy sounds better, see the `STARTUPINFOW` struct (https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/ns-processthreadsapi-startupinfow) and its `wShowWindow` member, which could probably be set to `SW_SHOWNOACTIVATE`
> Displays a window in its most recent size and position. This value is similar to `SW_SHOWNORMAL`, except that the window is not activated.

or to `SW_SHOWMINNOACTIVE`
> Displays the window as a minimized window. This value is similar to `SW_SHOWMINIMIZED`, except the window is not activated.

(https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-showwindow).

**Comment by Robyt3** [comment]
We have the `-s`/`--silent` command line flag, which completely hides the console window. Do we really need another flag to start minimized? Users could just create a shortcut to do the same thing.

**Comment by heinrich5991** [comment]
The problem is that starting the server from the client defocuses the client. I don't think we need a parameter for this, we can simply™ start the process in a way that doesn't make the terminal pop up.

**Comment by Robyt3** [comment]
We are already using `SW_SHOWMINNOACTIVE`

https://github.com/ddnet/ddnet/blob/f6febca706899e0d7a46f312460a74eaf08a7abe/src/base/system.cpp#L3783-L3793

**Comment by Robyt3** [comment]
Works for me.
